ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/validation/RequestValidator.java

```diff
@@ -180,4 +183,46 @@ private int validateRunScript(
     return validate(serviceContext, executionContext.parse(sql), statement.getOverrides(), sql);
   }
 
+  /**
+   * Performs permissions checks on the statement topics.
+   * </p>
+   * This check verifies the User and the KSQL server principal have the right ACLs permissions
+   * to access the statement topics.
+   *
+   * @param userServiceContext The context of the user executing this command.
+   * @param executionContext The execution context which contains the KSQL service context
+   *                         and the KSQL metastore.
+   * @param statement  The statement that needs to be checked.
+   */
+  private void validateTopicPermissions(
+      final ServiceContext userServiceContext,
+      final MetaStore metaStore,
+      final ConfiguredStatement<?> configuredStatement
+  ) {
+    final Statement statement = configuredStatement.getStatement();
+
+    topicAccessValidator.validate(userServiceContext, metaStore, statement);
+
+    // If these service contexts are different, then KSQL is running in a secured environment
+    // with authentication and impersonation enabled.
+    if (userServiceContext != serverServiceContext) {
+      try {
+        // Perform a permission check for the KSQL server
+        topicAccessValidator.validate(serverServiceContext, metaStore, statement);
+      } catch (final KsqlTopicAuthorizationException e) {
+        throw new KsqlStatementException(
+            "The KSQL service principal is not authorized to execute the command: "
+                + e.getMessage(),
+            configuredStatement.getStatementText()
+        );
```

Comment:
I had to append the `e.getMessage()` here because KSQL does not print the root cause message when a  `KsqlStatementException` is detected. I didn't want to fix that because I don't know the reason of not including the whole stack. Better fix it later.